src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -2040,7 +2040,7 @@ public ListBoxModel doFillConnectionStrategyItems(@QueryParameter String connect
 
         public FormValidation doCheckConnectionStrategy(@QueryParameter String connectionStrategy) {
             return Stream.of(ConnectionStrategy.values())
-                    .filter(v -> v.equalsName(connectionStrategy))
+                    .filter(v -> v.name().equals(connectionStrategy))
```

Comment:
Previously: Public_IP == PUBLIC_IP ??
Now:        PUBLIC_IP == PUBLIC_IP ??

Reply:
is it not the same? why is it fixing the error?

https://github.com/jenkinsci/ec2-plugin/blob/2f348aa519338aac64dd19954b0b41a62a8ea6fc/src/main/java/hudson/plugins/ec2/ConnectionStrategy.java#L15-L17

Reply:
I too am interested in Francisco's point

Reply:
`equalsName` is comparing the text added in the constructor (`Public IP` lowercase), but the `connectionStrategy` variable contains [the declared name of the enum](https://docs.oracle.com/javase/7/docs/api/java/lang/Enum.html#name()) (`PUBLIC IP` uppercase), so they were never equals.

Reply:
Oh! I see now! Really good point Ramon

Reply:
Interesting, then shouldn't `equals()` also produce the desired behaviour?